osbs/build/build_request.py

```diff
@@ -377,16 +377,32 @@ def adjust_for_registry_api_versions(self):
     def adjust_for_triggers(self):
         """
         Remove trigger-related plugins if no triggers set
+        import_image plugin is only removed for non-custom base images
+        Triggers are explicitly removed for custom base images
         """
         triggers = self.template['spec'].get('triggers', [])
+
+        remove_plugins = [
+            ("prebuild_plugins", "check_and_set_rebuild"),
+            ("prebuild_plugins", "stop_autorebuild_if_disabled"),
+            ("prebuild_plugins", "update_parent_image_stream_tag"),
+            ("exit_plugins", "sendmail"),
+        ]
+
+        should_remove = False
+        if triggers and self.is_custom_base_image():
+            msg = "removing %s from request because custom base image"
+            del self.template['spec']['triggers']
+            should_remove = True
+
         if not triggers:
```

Comment:
Could we make this `elif not triggers:` to make it clear that we don't ever want both blocks to run?